newsroom/search.py

```diff
@@ -639,6 +654,43 @@ def apply_request_filter(self, search):
             if search.args.get("created_from") or search.args.get("created_to"):
                 search.source["post_filter"]["bool"]["filter"].append(self.versioncreated_range(search.args))
 
+    def apply_request_advanced_search(self, search: SearchQuery):
+        if not search.args.get("advanced_search"):
+            return
+
+        search.advanced = json.loads(search.args["advanced_search"])
+        fields = search.advanced.get("fields") or self.default_advanced_search_fields
+        if not fields:
+            return
+
+        def gen_match_query(keywords: str, operator: str, multi_match_type):
+            if len(fields) == 1:
```

Comment:
just wondering if that multi match wouldn't do the same thing, is this part needed?

Reply:
You're right, under the hood it probably creates the same query. I'll upload a fix shortly